core/src/hlist.rs

```diff
@@ -1123,6 +1142,8 @@ impl HZippable<HNil> for HNil {
 impl<H1, T1, H2, T2> HZippable<HCons<H2, T2>> for HCons<H1, T1>
 where
     T1: HZippable<T2>,
+    <<T1 as HZippable<T2>>::Zipped as HList>::Len: Add<UInt<UTerm, B1>>,
+    <<<T1 as HZippable<T2>>::Zipped as HList>::Len as Add<UInt<UTerm, B1>>>::Output: Unsigned,
```

Comment:
a) we need to pull out the type-defined length of the post-zipped list, and ensure that it is a type that implements the Add trait, effectively "the type cat play the role of `a` in `a + 1`"
b) with that addition constaint satisfied, we need to get the output, and ensure that it is a type that can fill the roll of `b` in `a + 1 = b`